lib/draper/decoratable.rb

```diff
@@ -12,8 +12,14 @@ module Decoratable
     extend ActiveSupport::Concern
     include Draper::Decoratable::Equality
 
+    autoload :CollectionProxy, 'draper/decoratable/collection_proxy'
+
     included do
       prepend Draper::Compatibility::Broadcastable if defined? Turbo::Broadcastable
+
+      if defined? ActiveRecord
+        ActiveRecord::Associations::CollectionProxy.include CollectionProxy
```

Comment:
This file is loaded when ORM is Mongoid too.  So, I think a loading file only for Active Record isn't good here.
How about loading a file in `railtie.rb` like the following. 

```diff 
diff --git a/lib/draper/compatibility/active_record/collection_proxy.rb b/lib/draper/compatibility/active_record/collection_proxy.rb
new file mode 100644
index 0000000..f1ea1e2
--- /dev/null
+++ b/lib/draper/compatibility/active_record/collection_proxy.rb
@@ -0,0 +1,17 @@
+module Draper
+  module Compatibility
+    module ActiveRecord
+      module CollectionProxy
+        # Decorates a collection of objects. Used at the end of a scope chain.
+        #
+        # @example
+        #   company.products.popular.decorate
+        # @param [Hash] options
+        #   see {Decorator.decorate_collection}.
+        def decorate(options = {})
+          decorator_class.decorate_collection(load_target, options.reverse_merge(with: nil))
+        end
+      end
+    end
+  end
+end
diff --git a/lib/draper/railtie.rb b/lib/draper/railtie.rb
index 556c15a..78f1dfd 100644
--- a/lib/draper/railtie.rb
+++ b/lib/draper/railtie.rb
@@ -36,6 +36,11 @@ module Draper
       [:active_record, :mongoid].each do |orm|
         ActiveSupport.on_load orm do
           Draper.setup_orm self
+
+          if orm == :active_record
+            require 'draper/compatibility/active_record/collection_proxy'
+            ActiveRecord::Associations::CollectionProxy.include Draper::Compatibility::ActiveRecord::CollectionProxy
+          end
         end
       end
     end
```

Reply:
> This file is loaded when ORM is Mongoid too.

That's why `if defined? ActiveRecord` is here. It's similar to `if defined? Turbo::Broadcastable` a line before.

> So, I think a loading file only for Active Record isn't good here.

I'm sorry, but I don't get the point 🤔 `decoratable` isn't loaded just for AR, but `collection_proxy` is, as it is in your example too.

Logically, it's a part of `setup_orm` too. That's why I didn't put into the initializer directly.

Reply:
I'm not sure the philosophy of Draper correctly yet, but I think Draper tries to use the same codes for all ORM because there are no `defined?(ActiveRecord)`  and `defined?(Mongoid)`. I think we should follow it and add `defined?(ActiveRecord)` would against it.

So I think `lib/draper/railtie.rb` is a good place to add that because that's the only place to consider the ORM now.

Reply:
> I think `lib/draper/railtie.rb` is a good place to add that

Why not `setup_orm` then?

Reply:
`setup_orm` only knows a base class, but this PR needs to extend another class(doesn't touch a base class). So I think `setup_orm` isn't a good place to do it.
Or, adding another setup method like `setup_active_record` or `active_record_integration` maybe better because we already have `setup_action_controller` and `minitest_integration`(`rspec_integration`).

Reply:
> `setup_orm` only knows a base class, but this PR needs to extend another class(doesn't touch a base class).

So what? This still could be use to specify *dependencies* of that class.

> So I think `setup_orm` isn't a good place to do it.

Let's say it again: `setup_orm` isn't a good place to set up Active Record ORM.

> Or, adding another setup method like `setup_active_record` or `active_record_integration` maybe better because we already have `setup_action_controller` and `minitest_integration`(`rspec_integration`).

We have a lot of other mess and inconsistencies I'd like to refactor. But for now, I'd like to fix all the bugs in a backlog. I'm fine if someone does it in theirs own way. I'm afraid, having discussions on style in every PR is too much for me, sorry.

Reply:
Hmm. That isn't a minor thing for me. It's essential to keep code clean. But, I understand your opinion. I don't decline this. 

By the way, do you have a plan to add specs to this PR? I would like to keep adding specs for PRs that change behavior.

Reply:
> It's essential to keep code clean.

It's essential for me either. But it's already far from being clean. So, I try to fix bugs first and do my best in keeping off refactoring yet, minimizing diffs. Otherwise, I'd refactor initializers as well.

> By the way, do you have a plan to add specs to this PR?

Yes, sure. I've already reacted 👍 for that. Nevertheless of will you approve the PR or not, I'll make it complete.

Reply:
> `Draper::Compatibility::ActiveRecord::CollectionProxy`

It's not a _compatibility_, but our business logic.